### PR TITLE
FirebaseInstallations: more unit tests to prove correct behaviour

### PR DESCRIPTION
**Pull request by maksymmalyhin**

### CODE REVIEW

File: FirebaseInstallations/Source/Tests/Unit/FIRInstallationsIDControllerTests.m [hunk 1]
```diff
@@ -330,15 +331,61 @@ - (void)testGetInstallationItem_WhenThereIsIIDAndNoFID_ThenFIDIsCreatedAndRegist
...
+- (void)testGetInstallationItem_WhenCalledSeveralTimesWaitingForStore_OnlyOneOperationIsPerformed {
   // 1. Expect the installation to be requested from the store only once.
   FIRInstallationsItem *storedInstallation1 =
       [FIRInstallationsItem createRegisteredInstallationItem];
   FBLPromise<FIRInstallationsItem *> *pendingStorePromise = [FBLPromise pendingPromise];
   OCMExpect([self.mockInstallationsStore installationForAppID:self.appID appName:self.appName])
       .andReturn(pendingStorePromise);
 
-  // 3. Request installation n times
+  // 2. Request installation n times
```
Comment (paulb777) [inline FirebaseInstallations/Source/Tests/Unit/FIRInstallationsIDControllerTests.m]: Also the numbers below.

File: FirebaseInstallations/Source/Tests/Unit/FIRInstallationsIDControllerTests.m [hunk 2]
```diff
@@ -375,6 +422,45 @@ - (void)testGetInstallationItem_WhenCalledSeveralTimes_OnlyOneOperationIsPerform
...
+      XCTAssertNotNil(value);
+      XCTAssertEqualObjects(value.firebaseInstallationID,
+                            storedInstallation.firebaseInstallationID);
+      [getFIDExpectation fulfill];
+    }];
+
+    [self waitForExpectations:@[ getFIDExpectation ] timeout:0.5];
+  }
+
+  //   3. Finish API request.
```
Comment (paulb777) [inline FirebaseInstallations/Source/Tests/Unit/FIRInstallationsIDControllerTests.m]: inconsistent indentation

File: FirebaseInstallations/Source/Tests/Unit/FIRInstallationsIDControllerTests.m (hunk 1 above)
Comment (maksymmalyhin) [reply FirebaseInstallations/Source/Tests/Unit/FIRInstallationsIDControllerTests.m]: good catch! 😄

**Comment by paulb777** [comment]
Fix travis failure:

 - NOTE  | [iOS] xcodebuild:  Jan  7 18:25:38  AppHost-FirebaseInstallations-Unit-Tests[12469] <Error>: 6.14.0 - [Firebase/Installations][I-FIS002003] Firebase Installation registration failed for app with name: testInstallationsWithApp2, error: Error Domain=com.firebase.installations Code=0 "The server responded with an error. HTTP response: <NSHTTPURLResponse: 0x7ae5c4b0> { URL: https://firebaseinstallations.googleapis.com/v1/projects/(null)/installations/ } { status code: 403, headers {

**Comment by maksymmalyhin** [comment]
Response to [the comment](https://github.com/firebase/firebase-ios-sdk/pull/4612#issuecomment-571728364):

It is an expected error message for a misconfigured Firebase project which doesn't lead to a test failure.

**Comment by paulb777** [comment]
Then what is the reason for the failures in https://travis-ci.org/firebase/firebase-ios-sdk/jobs/633867182?

**Comment by maksymmalyhin** [comment]
@paulb777 I pushed the fix (at least that was the issue for my local tests on the old simulator)

**Pull request merged**